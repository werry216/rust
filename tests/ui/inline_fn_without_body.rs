// Copyright 2014-2018 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.





#![warn(clippy::inline_fn_without_body)]
#![allow(clippy::inline_always)]

trait Foo {
    #[inline]
    fn default_inline();

    #[inline(always)]fn always_inline();

    #[inline(never)]
    fn never_inline();

    #[inline]
    fn has_body() {
    }
}

fn main() {
}
