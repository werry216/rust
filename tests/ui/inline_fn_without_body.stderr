error: use of `#[inline]` on trait method `default_inline` which has no body
  --> $DIR/inline_fn_without_body.rs:18:5
   |
18 |       #[inline]
   |  _____-^^^^^^^^
19 | |     fn default_inline();
   | |____- help: remove
   |
   = note: `-D clippy::inline-fn-without-body` implied by `-D warnings`

error: use of `#[inline]` on trait method `always_inline` which has no body
  --> $DIR/inline_fn_without_body.rs:21:5
   |
21 |     #[inline(always)]fn always_inline();
   |     ^^^^^^^^^^^^^^^^^ help: remove

error: use of `#[inline]` on trait method `never_inline` which has no body
  --> $DIR/inline_fn_without_body.rs:23:5
   |
23 |       #[inline(never)]
   |  _____-^^^^^^^^^^^^^^^
24 | |     fn never_inline();
   | |____- help: remove

error: aborting due to 3 previous errors

